sdk/core/azure-core/inc/azure/core/match_conditions.hpp

```diff
@@ -19,6 +19,8 @@ namespace Azure {
  */
 struct MatchConditions
 {
+  virtual ~MatchConditions() = default;
```

Comment:
Same here/elsewhere - add doc comments.

Also, please avoid making such changes in a PR that is not related (do it separately). The intent/single purpose of this PR was `Add final specifier`.

Can we add a unit test for this to justify why we needed these virtual dtors?